mysql-test/columnstore/basic/t/union.test

```diff
@@ -15,48 +15,253 @@ USE mcs_union;
-CREATE TABLE t2 (a INT UNSIGNED);
+CREATE TABLE t2 (a MEDIUMINT UNSIGNED);
 INSERT INTO t2 VALUES (0),(1),(0xFFFFFF-2);
-(SELECT a FROM t1 UNION ALL SELECT a FROM t2) ORDER BY a;
-(SELECT a FROM t2 UNION ALL SELECT a FROM t1) ORDER BY a;
+SELECT * FROM (SELECT a FROM t1 UNION ALL SELECT a FROM t2) tu ORDER BY a;
+SELECT * FROM (SELECT a FROM t2 UNION ALL SELECT a FROM t1) tu ORDER BY a;
 DROP TABLE t1,t2;
 
 CREATE TABLE t1 (a INT);
 INSERT INTO t1 VALUES (CAST(0xFFFFFFFF80000000 AS SIGNED)+2),(-1),(0),(1),(0x7FFFFFFF);
 CREATE TABLE t2 (a INT UNSIGNED);
 INSERT INTO t2 VALUES (0),(1),(0xFFFFFFFF-2);
-(SELECT a FROM t1 UNION ALL SELECT a FROM t2) ORDER BY a;
-(SELECT a FROM t2 UNION ALL SELECT a FROM t1) ORDER BY a;
+SELECT * FROM (SELECT a FROM t1 UNION ALL SELECT a FROM t2) tu ORDER BY a;
+SELECT * FROM (SELECT a FROM t2 UNION ALL SELECT a FROM t1) tu ORDER BY a;
 DROP TABLE t1,t2;
 
 CREATE TABLE t1 (a BIGINT);
 INSERT INTO t1 VALUES (CAST(0x8000000000000000 AS SIGNED)+2),(-1),(0),(1),(0x7FFFFFFFFFFFFFFF);
 CREATE TABLE t2 (a BIGINT UNSIGNED);
 INSERT INTO t2 VALUES (0),(1),(0xFFFFFFFFFFFFFFFF-2);
-(SELECT a FROM t1 UNION ALL SELECT a FROM t2) ORDER BY a;
-(SELECT a FROM t2 UNION ALL SELECT a FROM t1) ORDER BY a;
+SELECT * FROM (SELECT a FROM t1 UNION ALL SELECT a FROM t2) tu ORDER BY a;
+SELECT * FROM (SELECT a FROM t2 UNION ALL SELECT a FROM t1) tu ORDER BY a;
+DROP TABLE t1,t2;
+
+--echo #
+--echo # Different Size Different Signedness Int Union
+--echo #
+
+CREATE TABLE t1 (a TINYINT);
+INSERT INTO t1 VALUES (CAST(0xFFFFFFFFFFFFFF80 AS SIGNED)+2),(-1),(0),(1),(0x7F);
+CREATE TABLE t2 (a INT UNSIGNED);
+INSERT INTO t2 VALUES (0),(1),(0xFFFF-2);
+SELECT * FROM (SELECT a FROM t1 UNION ALL SELECT * FROM t2) tu ORDER BY a;
+SELECT * FROM (SELECT a FROM t2 UNION ALL SELECT a FROM t1) tu ORDER BY a;
+DROP TABLE t1,t2;
+
+CREATE TABLE t1 (a SMALLINT);
+INSERT INTO t1 VALUES (CAST(0xFFFFFFFFFFFF8000 AS SIGNED)+2),(-1),(0),(1),(0x7FFF);
+CREATE TABLE t2 (a INT UNSIGNED);
+INSERT INTO t2 VALUES (0),(1),(0xFFFF-2);
+SELECT * FROM (SELECT a FROM t1 UNION ALL SELECT * FROM t2) tu ORDER BY a;
+SELECT * FROM (SELECT a FROM t2 UNION ALL SELECT a FROM t1) tu ORDER BY a;
+DROP TABLE t1,t2;
+
+CREATE TABLE t1 (a MEDIUMINT);
+INSERT INTO t1 VALUES (CAST(0xFFFFFFFFFF800000 AS SIGNED)+2),(-1),(0),(1),(0x7FFFFF);
+CREATE TABLE t2 (a INT UNSIGNED);
+INSERT INTO t2 VALUES (0),(1),(0xFFFFFF-2);
+SELECT * FROM (SELECT a FROM t1 UNION ALL SELECT a FROM t2) tu ORDER BY a;
+SELECT * FROM (SELECT a FROM t2 UNION ALL SELECT a FROM t1) tu ORDER BY a;
+DROP TABLE t1,t2;
+
+CREATE TABLE t1 (a BIGINT);
+INSERT INTO t1 VALUES (CAST(0x8000000000000000 AS SIGNED)+2),(-1),(0),(1),(0x7FFFFFFFFFFFFFFF);
+CREATE TABLE t2 (a INT UNSIGNED);
+INSERT INTO t2 VALUES (0),(1),(0xFFFFFF-2);
+SELECT * FROM (SELECT a FROM t1 UNION ALL SELECT a FROM t2) tu ORDER BY a;
+SELECT * FROM (SELECT a FROM t2 UNION ALL SELECT a FROM t1) tu ORDER BY a;
```

Comment:
We should also add DECIMAL test cases:
- Both sides DECIMAL with same signedness and precision.
- Both sides DECIMAL with same signedness and different precision.
- Both sides DECIMAL with different signedness and same precision.
- Both sides DECIMAL with different signedness and different precision.
- One side DECIMAL and other side integer type. Again, there should be variation in test cases in DECIMAL precision, integer types, and signedness.